### ISSUE TITLE
Large Meshes Miss Faces

### ISSUE THREAD

**Issue by digumx** [digumx/ProjectReticle#1]
Loading meshes with a large number of vertices and triangles (exact limit not confirmed) causes some faces to not be rendered. As an example, an archive of an example file that causes this to happen, and screenshots of the problem as well as the ideal case, as seen inside the assimp_viewer, is attached:

[Model]()

Expected:
![expected](https://user-images.githubusercontent.com/29953130/28084472-58c5d738-6696-11e7-8101-f9db27ce4455.PNG "Expected")

Problem:
![problem](https://user-images.githubusercontent.com/29953130/28084478-5ac45028-6696-11e7-8d2d-25a018d735b1.PNG "Problem")

Notice that though we are looking at the keyboard from the bottom (the model is of an Apple iMac keyboard), the sides of the keycaps are showing through. It's almost as if the entire bottom plane is missing.

Note: Don't mind the texturing and shading differences, those are supposed to happen.

Some thoughts on this:

 -  I've checked that the correct number of vertices and triangles are being loaded all along the pipeline. This, combined with some googling, makes me think that the problem is due to pushing so many triangles into the buffer that the integral type responsible for indexing is overflowing. The problem with this is that I am using GLuint, and am theoretically far from overflow.

**Comment by digumx** [comment]
Further investigation makes it appear as if index overflow isn't an issue. A call to `GL_MAX_ELEMENTS_VERITCES` and `GL_MAX_ELEMENTS_INDICES` are both 1048576, while the mesh loads 259740 vertex data buffer elements, and 51948 triangle data buffer elements.

**Comment by digumx** [comment]
It seems there was a problem with the .3ds file. I imported it into Blender, and immediately noticed that the component meshes had bad transforms. So, I manually fixed it and re-exported, and now the 3ds file opens nicely in Reticle.

So, it should seem that the issue is with having transforms assigned to the mesh components - niether Blender nor Reticle could apply these transforms correctly. However, assimp_viewer seems to be applying these correctly.
